src/Symfony/Component/Config/ResourceCheckerConfigCache.php

```diff
@@ -161,4 +143,35 @@ private function getMetaFile()
     {
         return $this->file.'.meta';
     }
+
+    private function safelyUnserialize($file)
+    {
+        $e = null;
+        $meta = false;
+        $signalingException = new \UnexpectedValueException();
+        $prevUnserializeHandler = ini_set('unserialize_callback_func', '');
+        $prevErrorHandler = set_error_handler(function ($type, $msg, $file, $line, $context) use (&$prevErrorHandler, $signalingException) {
+            if (E_WARNING === $type && 'Class __PHP_Incomplete_Class has no unserializer' === $msg) {
+                throw $signalingException;
+            }
+
+            return $prevErrorHandler ? $prevErrorHandler($type, $msg, $file, $line, $context) : false;
+        });
+
+        try {
+            $meta = unserialize(file_get_contents($file));
+        } catch (\Error $e) {
+        } catch (\Exception $e) {
+        }
+        restore_error_handler();
+        ini_set('unserialize_callback_func', $prevUnserializeHandler);
+        if (null !== $e && $e !== $signalingException) {
+            throw $e;
+        }
+        if (false === $meta) {
+            return false;
```

Comment:
Good catch! I just extracted the method without additional scrutinity and indeed these lines are no longer needed.